pkg/sql/tablemetadatacache/table_metadata_updater.go

```diff
@@ -15,55 +15,106 @@ import (
 	"context"
 	gojson "encoding/json"
 	"fmt"
+	"math"
 
+	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 )
 
 const pruneBatchSize = 512
 
+// iTableMetadataUpdater is an interface that exposes a RunUpdater
+// to be tableMetadataUpdateJobResumer. This interface primarily
+// serves as a way to facilitate better testing of tableMetadataUpdater
+// and tableMetadataUpdateJobResumer.
+type iTableMetadataUpdater interface {
+	RunUpdater(ctx context.Context) error
+}
+
 // tableMetadataUpdater encapsulates the logic for updating the table metadata cache.
 type tableMetadataUpdater struct {
-	ie isql.Executor
-	// upsertQuery is the query used to upsert table metadata rows,
-	// it is reused for each batch to avoid allocations between batches.
-	upsertQuery *tableMetadataBatchUpsertQuery
+	ie             isql.Executor
+	metrics        *TableMetadataUpdateJobMetrics
+	updateProgress func(ctx context.Context, progress float32)
 }
 
-// newTableMetadataUpdater creates a new tableMetadataUpdater.
-func newTableMetadataUpdater(ie isql.Executor) *tableMetadataUpdater {
-	return &tableMetadataUpdater{ie: ie, upsertQuery: newTableMetadataBatchUpsertQuery(tableBatchSize)}
+var _ iTableMetadataUpdater = &tableMetadataUpdater{}
+
+// newTableMetadataUpdater creates a new iTableMetadataUpdater.
+var newTableMetadataUpdater = func(job *jobs.Job, metrics *TableMetadataUpdateJobMetrics, ie isql.Executor) iTableMetadataUpdater {
```

Comment:
Can we continue to return the concrete type here? I believe it's typically preferred over returning the interface type.

Reply:
Done.